wrapper/src/gnatcuda_wrapper.adb

```diff
@@ -205,8 +216,29 @@ function GNATCUDA_Wrapper return Integer is
       return GNAT.OS_Lib.Spawn (S, Args);
    end Spawn;
 
+   ----------------------
+   -- Locate_And_Check --
+   ----------------------
+
+   function Locate_And_Check (Name : String) return String_Access is
+      Tmp : String_Access;
+   begin
+      Tmp := Locate_Exec_On_Path (Name);
+
+      if Tmp = null then
+         Put_Line ("error: could not locate """ & Name & """ on PATH");
+         raise Exec_Not_Found;
+      end if;
+
+      return Tmp;
+   end Locate_And_Check;
+
    --  Start of processing for GNATCCG_Wrapper
 
+   CUDA_Bin : constant String := GNAT.Directory_Operations.Dir_Name
+     (Locate_And_Check ("ptxas").all);
+   CUDA_Root : constant String := GNAT.Directory_Operations.Dir_Name
+     (CUDA_Bin (CUDA_Bin'First .. CUDA_Bin'Last - 1));
```

Comment:
Dir_Name of the file path minus its last character?

Reply:
Yes - last character is the directory separator.

Reply:
Ah yes I forgot how crusty the lib is.